src/xiana/swagger.clj

```diff
@@ -0,0 +1,47 @@
+(ns xiana.swagger
+  (:require
+    [jsonista.core :as json]
+    [xiana.route.helpers :as helpers]))
+
+(defn routes->swagger-json [routes
+                            & {type :type
+                               render? :render?
+                               route-opt-map :route-opt-map}]
+  (-> routes
+      helpers/routes->routes'
+      ((if (not (false? render?))
+         #(helpers/routes->swagger-data % :route-opt-map route-opt-map)
+         identity))
+      ((cond
+         (= type :json) json/write-value-as-string
+         (= type :edn) identity
+         :else identity))))
```

Comment:
https://clojuredocs.org/clojure.core/as-%3E

Reply:
That's not what I was meant to write actually. If you don't have any strong opinions, I would rather go with this way on that.